File: Other files/Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<int>>mp;
        int n = strs.size();
        for(int i=0;i<n;i++){
            string str = strs[i];
            sort(str.begin(),str.end());
            mp[str].push_back(i);
        }
        vector<vector<string>>ans;
        map<string,vector<int>> :: iterator itr;
        // cout<<mp.size();
        for(itr = mp.begin(); itr!=mp.end(); itr++){
            vector<int> arr = itr->second;
            // cout<<arr.size()<<" ";
            vector<string>temp;
            for(int i=0;i<arr.size(); i++){
                temp.push_back(strs[arr[i]]);
            }
            ans.push_back(temp);
        }
        return ans;
    }
};

File: GeekforGeeks Questions/DSA/groupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        strs.sort()
        d=[]
        for i in strs:
            d.append("".join(sorted(i)))
        d1={}
        for i in range(len(d)):
            if d[i] not in d1:
                d1[d[i]]=[strs[i]]
            else:
                d1[d[i]].append(strs[i])
        return d1.values()
        